scripts/autoreduction/reduce_EQSANS.py

```diff
@@ -199,35 +291,261 @@ def reduce_non_sample(events: EventWorkspace):
+    """
+    # The autoreduction service will pass AUTOREDUCE_IPTS_DIR as output directory.
+    # We add run number as subdir, otherwise there will be too many files under AUTOREDUCE_IPTS_DIR
+    run_number = str(events.getRunNumber())  # e.g. "105584"
+    ipts = SampleLogs(events).experiment_identifier.value[5:]  # e.g. "12345" when having IPTS-12345
+    reduced_files_dir = output_dir
+    if output_dir == AUTOREDUCE_IPTS_DIR.format(ipts=ipts):
+        reduced_files_dir = os.path.join(output_dir, run_number)
+
+    # find most appropriate reduction options and amend if necessary
+    amendment = {}
+    #  Example: /SNS/EQSANS/IPTS-12345/shared/autoreduce/reduction_options_105584.json
+    reduction_options_path = os.path.join(output_dir, f"reduction_options_{run_number}.json")
+    if os.path.exists(reduction_options_path) is False:
+        #  Example: /SNS/EQSANS/IPTS-12345/shared/autoreduce/reduction_options.json
+        amendment = {
+            "iptsNumber": ipts,
+            "sample": {"runNumber": run_number},
+            "outputFileName": f"EQSANS_{run_number}",  # prefix for all output files
+            "configuration": {"outputDir": reduced_files_dir},
+        }
+        reduction_options_path = os.path.join(AUTOREDUCE_IPTS_DIR.format(ipts=ipts), "reduction_options.json")
+        if os.path.exists(reduction_options_path) is False:
+            #  Fallback: /SNS/EQSANS/shared/autoreduce/reduction_options.json
+            reduction_options_path = os.path.join(AUTOREDUCE_DIR, "reduction_options.json")
+            amendment["beamCenter"] = {"runNumber": run_number}
+    footer = (
+        "<b>Note:</b> the links below can't be open directly. Instead, copy the link address "
+        "and paste it in a new browser tab.<br>\n"
+    )
+    footer += "<table border='0'><tr><td>\n"
+    footer += f"Input reduction options loaded from</td><td>{filelink(reduction_options_path)}</td></tr>\n"
+    footer += "</table>\n"
+
+    # load options and validate
+    with open(reduction_options_path, "r") as f:
+        raw_options = json.load(f)
+    input_config = reduction_parameters(raw_options, validate=False, permissible=True)
+    input_config = update_reduction_parameters(input_config, amendment, validate=True, permissible=True)
+    final_input_config = deepcopy(input_config)  # input_config will be modified during reduction
+
+    # load files and reduce
+    loaded = load_all_files(input_config)
+    output = reduce_single_configuration(loaded, input_config)
+
+    # create plot images for all intensity profiles
+    plot_reduction_output(output, input_config)  # save as *.PNG files
+
+    # collect all HTML elements into a single report
+    report = reduce_non_sample(events) + "<hr>\n" + plotly_reduction_output(output, input_config) + "<hr>\n"
+
+    # Save the input reduction options
+    reduction_options_path = os.path.join(output_dir, f"reduction_options_{run_number}.json")
```

Comment:
Should this be in the run subdirectory rather than the IPTS autoreduce directory?
`/SNS/EQSANS/IPTS-12345/shared/autoreduce/105584/`
I guess it depends on whether we want to overwrite the input file from line 343 or not?
```suggestion
    reduction_options_path = os.path.join(reduced_files_dir, f"reduction_options_{run_number}.json")
```

Reply:
I thought about it but I believe it's best to leave it in `output_dir` because these are `input` configuration options, while `reduced_files_dir` contains `output` results. Perhaps more to the point, when autoreduction starts for a new run, the script will look for this particular file first. I believe it's easier to place this file in an already existing directory such as `output_dir`